jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -471,6 +471,7 @@ public static String decodePath(String path, int offset, int length)
                             if (u == 'u')
                             {
                                 // UTF16 encoding is only supported with UriCompliance.Violation.UTF16_ENCODINGS.
+                                // This is wrong. This is a codepoint not a char
```

Comment:
This needs to be fixed then? 9.4.x too?

Reply:
Nope - this is just how we implemented %u.  It is not strictly correct, but let's not improve a feature we really want to remove.